internal/stacks/stackruntime/plan_test.go

```diff
@@ -236,6 +237,241 @@ func TestPlanVariableOutputRoundtripNested(t *testing.T) {
 	}
 }
 
+var cmpCollectionsSet = cmp.Comparer(func(x, y collections.Set[stackaddrs.AbsComponent]) bool {
+	if x.Len() != y.Len() {
+		return false
+	}
+
+	for _, v := range x.Elems() {
+		if !y.Has(v) {
+			return false
+		}
+	}
+
+	return true
+})
```

Comment:
FWIW I an equivalent thing to package `collections` as part of https://github.com/hashicorp/terraform/pull/34483 before I realized you'd written this inline here.

Of course it's not a big deal to have a custom comparer here since this is just test code anyway, but if you'd like to consolidate then you could try using `collections.CmpOptions` instead of `cmpCollectionSet` in the tests below.

(In the final PR I ended up writing a transformer rather than a comparer, because that allows cmp to produce more specific diffs when there are mismatches.)